sim/scenarios/corrupt-rate/corrupt-rate-error-model.cc

```diff
@@ -30,39 +32,22 @@ void CorruptRateErrorModel::DoReset(void) { }
 bool CorruptRateErrorModel::DoCorrupt(Ptr<Packet> p) {
     if (distr(*rng) >= rate) return false;
 
-    const uint32_t p_len = p->GetSize();
-    uint8_t *buffer= new uint8_t[p_len];
-    p->CopyData(buffer, p_len);
-    PppHeader ppp_hdr = PppHeader();
-    uint32_t ppp_hdr_len = p->RemoveHeader(ppp_hdr);
-    Ipv4Header ip_hdr = Ipv4Header();
-    uint32_t ip_hdr_len = p->RemoveHeader(ip_hdr);
-    UdpHeader udp_hdr = UdpHeader();
-    uint32_t udp_hdr_len = p->RemoveHeader(udp_hdr);
-    const uint32_t hdr_len = ppp_hdr_len + ip_hdr_len + udp_hdr_len;
+    QuicPacket qp = QuicPacket(p);
+    // Don't corrupt Version Negotiation packets.
+    // Version Negotiation packets are expected to be sent when the version field of the Initial was corrupted.
+    // Client are supposed to ignore Version Negotiation packets that contain the version that they offered.
+    if(qp.IsVersionNegotiationPacket()) {
+        qp.ReassemblePacket();
+        return false;
```

Comment:
Why is this necessary? If the VN packet is corrupted, the client is likely to drop the packet, right?

Reply:
This requires 2 things to happen:
1. The version number field on the client's Initial is corrupted, eliciting a VN packet from the server.
2. The VN packet will now contain the (uncorrupted) version that the client selected. This tells the client that it can ignore the VN packet. But if this version number in the VN packet now is *also* corrupted, the client would accept the VN packet and the connection attempt will fail.

Reply:
@janaiyengar See https://github.com/quicwg/base-drafts/pull/3533.

Reply:
I think that's a good idea. Corrupted VN generate random failures, which distracts from the purpose of the test.

Reply:
I'll take this as an approval of this PR 🙃